### PR TITLE
feat(datastore) Allow configure DataStore with custom configuration

### PR DESCRIPTION
**Pull request by HuiSF**
Changes include iOS support only.
Waiting on `amplify-android` to release this [commit](https://github.com/aws-amplify/amplify-android/commit/0ae1319333bd1a9e7b6d0c722a255427f318573f#diff-1d6639ae83d38f3234fb312bbae544f6b9d92a34a84ce6d046920e29c7d73cd1) to add custom configuration in android

Custom configuration include
* `syncInterval`
* `syncMaxRecords`
* `syncPageSize`

### CODE REVIEW

File: packages/amplify_datastore_plugin_interface/lib/amplify_datastore_plugin_interface.dart [hunk 1]
```diff
@@ -44,16 +44,42 @@ export 'src/types/models/subscription_event.dart';
...
+  /// [syncPageSize]: page size to sync
   Future<void> configureDataStore(
-      {@required ModelProviderInterface modelProvider}) {
-    throw UnimplementedError('configureDataStore() has not been implemented.');
+      {@required ModelProviderInterface modelProvider,
+      double syncInterval,
+      int syncMaxRecords,
+      int syncPageSize}) {
+    throw UnimplementedError(
+        'configureModelProvider() has not been implemented.');
```
Comment (cshfang) [inline packages/amplify_datastore_plugin_interface/lib/amplify_datastore_plugin_interface.dart:82]: configureDataStore() has not been ..

**Comment by cshfang** [comment]
Will you do Android in a different commit?

**Comment by HuiSF** [comment]
> Will you do Android in a different commit?

Yes will add the android support in a different commit as the required API change hasn't been released within amplify-android package

**Comment by HuiSF** [comment]
Updated amplify-android to 1.17.6 which contains the required API change.
Updated PR to include custom configuration support for Android.

**Pull request closed without merging**